app/ide-desktop/lib/content/src/index.ts

```diff
@@ -148,75 +160,84 @@ class Main {
-                        if (
-                            !contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value
-                        ) {
-                            hideAuth()
-                            if (!appInstanceRan) {
-                                appInstanceRan = true
-                                void appInstance.run()
-                            }
-                        }
-                    }
-                    authentication.run({
-                        logger,
-                        platform,
-                        projectManager: projectManager.ProjectManager.default(),
-                        showDashboard:
-                            contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value,
-                        onAuthenticated,
-                    })
-                } else {
-                    void appInstance.run()
-                }
                 const email = contentConfig.OPTIONS.groups.authentication.options.email.value
                 // The default value is `""`, so a truthiness check is most appropriate here.
                 if (email) {
                     logger.log(`User identified as '${email}'.`)
                 }
+                void this.app.run()
+            }
+        }
+    }
+
+    main() {
+        contentConfig.OPTIONS.loadAll([app.urlParams()])
+        const isUsingAuthentication = contentConfig.OPTIONS.options.authentication.value
+        const isUsingNewDashboard =
+            contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value
+        const isOpeningIde =
+            contentConfig.OPTIONS.groups.startup.options.entry.value ===
+            contentConfig.OPTIONS.groups.startup.options.entry.default
+        if ((isUsingAuthentication || isUsingNewDashboard) && isOpeningIde) {
+            const hideAuth = () => {
+                const auth = document.getElementById('dashboard')
+                const ide = document.getElementById('root')
+                if (auth) {
+                    auth.style.display = 'none'
+                }
+                if (ide) {
+                    ide.hidden = false
+                }
+            }
+            /** This package is an Electron desktop app (i.e., not in the Cloud), so
+             * we're running on the desktop. */
+            /** TODO [NP]: https://github.com/enso-org/cloud-v2/issues/345
+             * `content` and `dashboard` packages **MUST BE MERGED INTO ONE**. The IDE
+             * should only have one entry point. Right now, we have two. One for the cloud
+             * and one for the desktop. Once these are merged, we can't hardcode the
+             * platform here, and need to detect it from the environment. */
+            const currentPlatform = contentConfig.OPTIONS.groups.startup.options.platform.value
+            const webPlatform = contentConfig.OPTIONS.groups.startup.options.platform.default
```

Comment:
This line is very tricky. It will stop working if we change the default there, which makes the code really hard to refactor. I think it would be better (still far from best solution known from real typed languages) to do `if (currentPlatform == 'web')`. This will work even if we change defaults.